beginner_source/nlp/advanced_tutorial.py

```diff
@@ -1,120 +1,112 @@
 ======================================================
 
-Dynamic versus Static Deep Learning Toolkits
+동적, 정적 딥 러닝 툴킷(toolkits) 비교
 --------------------------------------------
 
-Pytorch is a *dynamic* neural network kit. Another example of a dynamic
-kit is `Dynet <https://github.com/clab/dynet>`__ (I mention this because
-working with Pytorch and Dynet is similar. If you see an example in
-Dynet, it will probably help you implement it in Pytorch). The opposite
-is the *static* tool kit, which includes Theano, Keras, TensorFlow, etc.
-The core difference is the following:
-
-* In a static toolkit, you define
-  a computation graph once, compile it, and then stream instances to it.
-* In a dynamic toolkit, you define a computation graph *for each
-  instance*. It is never compiled and is executed on-the-fly
-
-Without a lot of experience, it is difficult to appreciate the
-difference. One example is to suppose we want to build a deep
-constituent parser. Suppose our model involves roughly the following
-steps:
-
-* We build the tree bottom up
-* Tag the root nodes (the words of the sentence)
-* From there, use a neural network and the embeddings
-  of the words to find combinations that form constituents. Whenever you
-  form a new constituent, use some sort of technique to get an embedding
-  of the constituent. In this case, our network architecture will depend
-  completely on the input sentence. In the sentence "The green cat
-  scratched the wall", at some point in the model, we will want to combine
-  the span :math:`(i,j,r) = (1, 3, \text{NP})` (that is, an NP constituent
-  spans word 1 to word 3, in this case "The green cat").
-
-However, another sentence might be "Somewhere, the big fat cat scratched
-the wall". In this sentence, we will want to form the constituent
-:math:`(2, 4, NP)` at some point. The constituents we will want to form
-will depend on the instance. If we just compile the computation graph
-once, as in a static toolkit, it will be exceptionally difficult or
-impossible to program this logic. In a dynamic toolkit though, there
-isn't just 1 pre-defined computation graph. There can be a new
-computation graph for each instance, so this problem goes away.
-
-Dynamic toolkits also have the advantage of being easier to debug and
-the code more closely resembling the host language (by that I mean that
-Pytorch and Dynet look more like actual Python code than Keras or
-Theano).
-
-Bi-LSTM Conditional Random Field Discussion
+Pytorch는 *동적* 신경망 툴킷입니다. 다른 동적인 신경망 툴킷 중에는
+`Dynet <https://github.com/clab/dynet>`_ 이 있습니다.(이 툴킷을
+쓰는 법이 Pytorch와 비슷하여 예시를 들었습니다. Dynet의 예제를 보면
+Pytorch로 구현할 때도 도움이 될 것입니디.) 반대로 *정적*인 툴킷들로
+Theano, Keras, TensorFlow 등이 있습니다. 주요 차이점은 다음과 같습니다:
+
+* 정적 툴킷을 쓸 때는 계산 그래프를 한 번만 정의하고, 컴파일 한 후,
+  데이터를 계산 그래프에 넘깁니다.
+* 동적 툴킷에서는 *각 데이터*의 계산 그래프를 정의하며 컴파일하지
+  않고 즉각 실행됩니다.
+
+경험이 많지 않다면 두 방식의 차이를 알기 어렵습니다. 딥 러닝 기반의
+구구조 분석기(constituent parser)를 예로 들어보겠습니다. 모델은 대략
+다음과 같은 과정을 수행합니다:
+
+* 트리를 상향식으로 만들어 나간다.
```

Comment:
```suggestion
* 트리를 상향식(bottom-up)으로 만들어 나간다.
```
상향식이라고 했을 때 bottom-up이 바로 떠오르지 않아 리뷰합니다.

Reply:
다른 분이 같은 줄에 다른 수정 제안을 해주셔서 겹치는 바람에 동시에 반영이 안된다고 하네요.
제안해주신대로 제가 따로 수정했습니다.